cdk/database/__init__.py

```diff
@@ -62,3 +67,27 @@ def dynamodb_single_table(self):
                                         partition_key=aws_dynamodb.Attribute(
                                             name='PK',
                                             type=aws_dynamodb.AttributeType.STRING))
+
+    def dynamodb_visits_table(self):
+        self.visits_table = aws_dynamodb.Table(self,
+                                               self.visits_id,
+                                               point_in_time_recovery=True,
+                                               removal_policy=core.RemovalPolicy.RETAIN,
+                                               sort_key=aws_dynamodb.Attribute(
+                                                   name='username',
+                                                   type=aws_dynamodb.AttributeType.STRING),
+                                               partition_key=aws_dynamodb.Attribute(
+                                                   name='visit_time',
+                                                   type=aws_dynamodb.AttributeType.STRING))
+
+    def dynamodb_users_table(self):
+        self.users_table = aws_dynamodb.Table(self,
+                                              self.users_id,
+                                              point_in_time_recovery=True,
+                                              removal_policy=core.RemovalPolicy.RETAIN,
+                                              sort_key=aws_dynamodb.Attribute(
+                                                  name='last_name',
```

Comment:
Out of curiosity, why did we choose last name as a sort key here? Sort keys are not a requirement for a table- they are just useful tools when you need a composite key to establish uniqueness. As an example- if this was going to be turned into a login system across many different universities- you might choose a key structure like: PK: school, SK: username. This would be because username would no longer be globally unique. It would just be unique to a partition key.

Reply:
At the time, I was just running on the assumption that sort key was neccesary.